sdk/cosmos/azure-cosmos/azure/cosmos/aio/_container.py

```diff
@@ -239,6 +235,17 @@ async def create_item(
         :returns: A CosmosDict representing the new item. The dict will be empty if `no_response` is specified.
         :rtype: ~azure.cosmos.CosmosDict[str, Any]
         """
+        etag = kwargs.pop('etag', None)
+        if etag is not None:
+            warnings.warn(
+                "The 'etag' flag does not apply to this method and will be removed in the future.",
+                UserWarning)
```

Comment:
I'd probably go with `DeprecationWarning`

Reply:
was just following the pattern we had for the deprecation of `populate_query_metrics` but I can go ahead and change them

Reply:
I think there's certain tooling implications for one vs the other - if someone has code running in a service, turning on filters for DeprecationWarning is a relatively standard practice - however I'm not sure what scenarios/industry recommendations for UserWarning might be, and where I would expect to filter for them/encounter them.
So could be worth double checking that before selecting the best option here.